pkg/operator/controller/status/controller.go

```diff
@@ -570,6 +610,68 @@ func computeGatewayAPICRDsDegradedCondition(state operatorState) configv1.Cluste
 	return degradedCondition
 }
 
+// computeGatewayAPIInstallDegradedCondition computes the degraded condition for the Gateway API OSSM subscription. It
+// checks for known conflicting subscriptions, as well as already existing subscription(s) to OSSM3, and reports
+// degraded when any of those subscriptions would prevent the installation of an appropriate Istio control plane.
```

Comment:
Can we make this conditional on having Gateway API enabled (that is, having a gatewayclass with our controller name)?

Reply:
It might be safer to check the gatewayclass, in case a conflicting subscription or some other issue is preventing the operator from creating the expected subscription.